docs/sources/flow/config-language/expressions/types_and_values.md

```diff
@@ -82,25 +89,53 @@ is on a different line as the final value:
 ```
 
 ## Objects
+
 Object values are constructed by a sequence of comma separated key-value pairs
 surrounded by curly braces `{}`:
-```
+
+```river
 {
   first_name = "John",
-  last_name = "Doe",
+  last_name  = "Doe",
 }
 ```
+
 A comma after the final key-value pair may be omitted if the closing curly
 brace `}` is on the same line as the final pair:
-```
+
+```river
 { name = "John" }
 ```
 
+If the key is not a valid identifier, it must be wrapped in double quotes like
+a string:
+
+```river
+{
+  "app.kubernetes.io/name"     = "mysql",
+  "app.kubernetes.io/instance" = "mysql-abcxyz",
+  namespace                    = "default",
+}
+```
+
+> **NOTE**: Be careful not to confuse objects with blocks.
+>
+> An _object_ is a value assigned to an [Attribute][Attributes], where
+> commas **must** be provided between key-value pairs on separate lines.
+>
+> A [Block][Blocks] is a structural element composed of multiple attributes,
```

Comment:
```suggestion
> A [Block][Blocks] is a named structural element composed of multiple attributes,
```

nit: does this make the difference any clearer? Not sure if it's worth it.

Reply:
sure, I like that